core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -166,7 +166,7 @@
       </ul>
     </li>
 
-    <li uib-dropdown ng-if="!username" class="oppia-navbar-clickable-dropdown">
+    <li uib-dropdown ng-if="!userIsLoggedIn && username != ''" class="oppia-navbar-clickable-dropdown">
```

Comment:
Maybe `!!username` is safer? Not sure if you're intending to handle cases like username being null/undefined here.